### PR TITLE
Adding check outliers

### PR DESCRIPTION
**Pull request by mhkhan27**

### CODE REVIEW

**Comment by mhkhan27** [comment]
[check for outliers](https://trello.com/c/J5opO5SE/133-check-for-outliers)

File: R/check_outliers.R [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#' @param minumum_unique_value_of_variable Default is NULL, mean this parameter won't be considered. For example 10 means for any variable where number of unique value is less than 10, then the variable won't be considered for outlier checking.
+#' @return Outliers
+#' @export
+#'
+
+
+check_outliers <- function(df,kobo_tool_location=NULL,
+                              columns_to_add,
+                              strongness_factor = 1.5,
+                              minumum_unique_value_of_variable =NULL,
```
Comment (yannsay-impact) [inline R/check_outliers.R:18]: typo in minumum

File: tests/testthat/test-check_outliers.R [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    expense = c(sample(200:500,replace = T,size = 95),c(600,100,80,1020,1050)),
+    income = c(sample(20000:50000,replace = T,size = 95),c(60,0,80,1020,1050))
+  )
+
+  outliers5 <- check_outliers(df = df_outlier,columns_to_add = "uuid",strongness_factor = 1.5)
+
+  boxplot_outlier_age <- (boxplot.stats(df_outlier$age))$out
+  outliers5_age_outliers <- (outliers5[outliers5$question == "age",])$old_value
+  expect_true(all(boxplot_outlier_age %in% outliers5_age_outliers ))
+
```
Comment (yannsay-impact) [inline tests/testthat/test-check_outliers.R:35]: not sure what line 30 to 34 are about. You are checking that the outliers from the IQR definition are present in the log but this is not what you are actually calculating. While the 3 values from boxplot_outlier_age are present, the outliers5 has 11 values for age.

**Review by yannsay-impact — CHANGES_REQUESTED** [review]

File: tests/testthat/test-check_outliers.R [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  outliers2 <- check_outliers(raw_data,columns_to_add = "X_uuid")
+
+  expect_false("enumerator_num" %in% outliers1$question)
+  expect_true("X_uuid" %in% names(outliers1))
+  expect_true("enumerator_num" %in% outliers2$question)
+
+  ## test with null in unique number
+  outliers3 <- check_outliers(raw_data,columns_to_add = "X_uuid",minumum_unique_value_of_variable = 10)
+  outliers4 <- check_outliers(raw_data,columns_to_add = "X_uuid",minumum_unique_value_of_variable = NULL)
+
```
[the remark this answers is not in the document]
Comment (mhkhan27) [reply tests/testthat/test-check_outliers.R:18]: Not sure! I am not having NAs as outliers.

File: R/check_outliers.R [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+
+    outliers_value_log <- log_variable[outliers_tf] %>% unique()
+
+
+
+    outlier_checks[[paste0("log_",x)]] <-  df %>% mutate(
+      issue = case_when(df[["log"]] %in%  outliers_value_log ~ "outlier (log distribution)"),
+    ) %>% filter(issue == "outlier (log distribution)") %>% select(all_of(columns_to_add),issue,all_of(x)) %>%
+      pivot_longer(cols = paste0(x),names_to ="question",values_to= "old_value")
```
[the remark this answers is not in the document]
Comment (mhkhan27) [reply R/check_outliers.R:126]: Lets keep **question** and **old_value**

**Pull request merged**